drivers/clock_control/clock_stm32g4.c

```diff
@@ -25,37 +24,69 @@
 #define z_pllr(v) LL_RCC_PLLR_DIV_ ## v
 #define pllr(v) z_pllr(v)
 
+/**
+ * @brief Return PLL source
+ */
+__unused
+static uint32_t get_pll_source(void)
+{
+	/* Configure PLL source */
+	if (IS_ENABLED(STM32_PLL_SRC_HSI)) {
+		return LL_RCC_PLLSOURCE_HSI;
+	} else if (IS_ENABLED(STM32_PLL_SRC_HSE)) {
+		return LL_RCC_PLLSOURCE_HSE;
+	}
+
+	return -ENOTSUP;
+}
+
 /**
  * @brief Set up pll configuration
  */
+__unused
 int config_pll_sysclock(void)
 {
-	uint32_t pll_source, pll_m, pll_n, pll_r;
+	uint32_t pll_source;
+
+	pll_source = get_pll_source();
+	if (pll_source < 0) {
+		return -ENOTSUP;
+	}
 
 	/* set power boost mode for sys clock greater than 150MHz */
 	if (sys_clock_hw_cycles_per_sec() >= MHZ(150)) {
 		LL_PWR_EnableRange1BoostMode();
 	}
 
-	pll_n = STM32_PLL_N_MULTIPLIER;
-	pll_m = pllm(STM32_PLL_M_DIVISOR);
-	pll_r = pllr(STM32_PLL_R_DIVISOR);
-
-	/* Configure PLL source */
-	if (IS_ENABLED(STM32_PLL_SRC_HSI)) {
-		pll_source = LL_RCC_PLLSOURCE_HSI;
-	} else if (IS_ENABLED(STM32_PLL_SRC_HSE)) {
-		pll_source = LL_RCC_PLLSOURCE_HSE;
-	} else {
-		return -ENOTSUP;
-	}
-
-	LL_RCC_PLL_ConfigDomain_SYS(pll_source, pll_m, pll_n, pll_r);
+	LL_RCC_PLL_ConfigDomain_SYS(pll_source,
+				    pllm(STM32_PLL_M_DIVISOR),
+				    STM32_PLL_N_MULTIPLIER,
+				    pllr(STM32_PLL_R_DIVISOR));
 
 	LL_RCC_PLL_EnableDomain_SYS();
 
 	return 0;
 }
+
+/**
+ * @brief Return pllout frequency
+ */
+__unused
+uint32_t get_pllout_frequency(void)
+{
+	uint32_t pll_source;
+
+	pll_source = get_pll_source();
```

Comment:
`get_pll_source()` returns `uint32_t`.
Therefore, `if (pll_source < 0)` is always `false`.